src/commcare_cloud/alias.py

```diff
@@ -24,9 +24,24 @@ def encode_string(value):
     return argv
 
 
+def _generate_args(*args, **kwargs):
+    argv = []
+
+    for arg in args:
+        argv.append(arg)
+    for key, value in kwargs.items():
+        if value is False or value is None:
+            continue
+        elif value is True:
+            argv.append('--{}'.format(key))
+        else:
+            argv.extend(['--{}'.format(key), value])
+    return argv
+
+
 def commcare_cloud(*args, **kwargs):
     from .cli_utils import print_command
     from .commcare_cloud import call_commcare_cloud
-    argv = _encode_args('commcare-cloud', *args, **kwargs)
+    argv = _generate_args('commcare-cloud', *args, **kwargs)
```

Comment:
@dannyroberts I made this change last week because it wasn't obvious to me why the arguments needed to be encoded for either 2 or 3 but maybe you can provide some additional context here?

Reply:
Interesting. Does this change work both in python 2 and 3? I think since `' '` in python 2 means `b' '` and `' '` in python 3 means `u' '`, the like least-change change would be to replace `' '.join(shlex_quote(arg) for arg in command)` with `b' '.join(shlex_quote(arg) for arg in command)`, but I'm honestly not sure whether the `sys.argv` and functions we use to run commands are expected to use list/tuples of `bytes` or of `unicode`. So I'm not sure whether the change you made here or the one I brought up is the more correct fix

Reply:
> I think since `' '` in python 2 means `b' '`

Except not in the module because of `from __future__ import unicode_literals`

I don't think we want to use bytes in Python 3, so `b' '.join(...)` seems wrong to me.

The issue @gherceg hit was due to printing, and I think will not be resolved by that change anyway.

https://github.com/dimagi/commcare-cloud/blob/d38f180f52f25219878cd170a10dce0211a76019/src/commcare_cloud/cli_utils.py#L96-L97

Reply:
Python docs on sys.argv - `Note On Unix, command line arguments are passed by bytes from OS. Python decodes them with filesystem encoding and “surrogateescape” error handler. When you need original bytes, you can get it by [os.fsencode(arg) for arg in sys.argv].`

This does work on python 2 (tested via staging deploy), as well as python 3.